usaspending_api/references/management/commands/load_rosetta.py

```diff
@@ -41,36 +41,33 @@ class Command(BaseCommand):
     s3_public_url = settings.DATA_DICTIONARY_DOWNLOAD_URL
 
     def add_arguments(self, parser):
-        parser.add_argument("-p", "--path", help="filepath to a local Excel spreadsheet to load", default=None)
+        parser.add_argument("-p", "--path", help="filepath to an Excel spreadsheet to load", default=self.s3_public_url)
 
     def handle(self, *args, **options):
+        script_start_time = perf_counter()
         logger.info("Starting load_rosetta management command")
+        logger.info("Loading data from {}".format(options["path"]))
 
-        script_start_time = perf_counter()
-        rosetta_object = extract_data_from_source_file(path=options["path"])
+        try:
+            local_filepath = Path("temp_local_data_dictionary.xlsx")
+            RetrieveFileFromUri(options["path"]).copy(str(local_filepath))
+            rosetta_object = extract_data_from_source_file(filepath=local_filepath)
+        except Exception:
+            logger.exception("Exception during file retrieval or parsing")
+        finally:
+            if local_filepath.exists():
+                local_filepath.unlink()
+
+        rosetta_object["metadata"]["download_location"] = options["path"]
 
-        if options["path"] is None:
-            rosetta_object["metadata"]["download_location"] = self.s3_public_url
         load_xlsx_data_to_model(rosetta_object)
 
         logger.info("Script completed in {:.2f}s".format(perf_counter() - script_start_time))
 
 
-def extract_data_from_source_file(path: str = None) -> dict:
-    if path:
-        logger.info("Using local file: {}".format(path))
-        filepath = path
-        download_path = None
-    else:
-        download_path = settings.ROSETTA_DICT_S3_PATH
-        bucket_name = download_path.split("/")[0]
-        filepath = "temp_data_act_crosswalk.xlsx"
-        logger.info("Using S3 file: {} from {}".format(filepath, bucket_name))
-        s3_client = boto3.client("s3", region_name=settings.USASPENDING_AWS_REGION)
-        s3_client.download_file(bucket_name, "/".join(settings.ROSETTA_DICT_S3_PATH.split("/")[1:]), filepath)
-
-    file_size = os.path.getsize(filepath)
-    wb = load_workbook(filename=filepath)
+def extract_data_from_source_file(filepath: str) -> dict:
```

Comment:
This filepath parameter used to be optional, and the `-p` arg from the command still is. Is there some reason we know a path will always be provided now?

Reply:
The default value for the argument was changed from None to the global value. So it will always be a value